src/main/java/org/opensearch/flowframework/transport/DeprovisionWorkflowTransportAction.java

```diff
@@ -151,64 +127,43 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
         }
     }
 
-    private void getResourcesAndExecute(
-        String workflowId,
-        List<ProcessNode> provisionProcessSequence,
-        ActionListener<WorkflowResponse> listener
-    ) {
-        GetWorkflowStateRequest getStateRequest = new GetWorkflowStateRequest(workflowId, true);
-        client.execute(GetWorkflowStateAction.INSTANCE, getStateRequest, ActionListener.wrap(response -> {
-            // Get a map of step id to created resources
-            final Map<String, ResourceCreated> resourceMap = response.getWorkflowState()
-                .resourcesCreated()
-                .stream()
-                .collect(Collectors.toMap(ResourceCreated::workflowStepId, Function.identity()));
-
-            // Now finally do the deprovision
-            executeDeprovisionSequence(workflowId, resourceMap, provisionProcessSequence, listener);
-        }, exception -> {
-            String message = "Failed to get workflow state for workflow " + workflowId;
-            logger.error(message, exception);
-            listener.onFailure(new FlowFrameworkException(message, ExceptionsHelper.status(exception)));
-        }));
-    }
-
     private void executeDeprovisionSequence(
         String workflowId,
         Map<String, ResourceCreated> resourceMap,
-        List<ProcessNode> provisionProcessSequence,
         ActionListener<WorkflowResponse> listener
     ) {
+
         // Create a list of ProcessNodes with the corresponding deprovision workflow steps
-        List<ProcessNode> deprovisionProcessSequence = provisionProcessSequence.stream()
-            // Only include nodes that created a resource
-            .filter(pn -> resourceMap.containsKey(pn.id()))
-            // Create a new ProcessNode with a deprovision step
-            .map(pn -> {
-                String stepName = pn.workflowStep().getName();
-                String deprovisionStep = getDeprovisionStepByWorkflowStep(stepName);
-                // Unimplemented steps presently return null, so skip
-                if (deprovisionStep == null) {
-                    return null;
-                }
-                // New ID is old ID with deprovision added
-                String deprovisionStepId = pn.id() + DEPROVISION_SUFFIX;
-                return new ProcessNode(
+        List<ProcessNode> deprovisionProcessSequence = new ArrayList<>();
+        for (Map.Entry<String, ResourceCreated> resource : resourceMap.entrySet()) {
+            String workflowStepId = resource.getKey();
+            ResourceCreated resourceCreated = resource.getValue();
```

Comment:
Actually on second look, I think you're losing a deprovision step here.   Your example in the comments shows the same step id and resource but different step names.  That results in only one entry in the map.  But you need to both undeploy and delete (unregister) the model ID.

Reply:
Ah I forgot to update the example in the description, but the step names are modified when the local/remote model registration also deploys. 

Here's an example from running the register remote model template with the deploy flag, the deploy step has a workflow step ID of `workflow_step_2-deploy` : 
```
{
  "workflow_id" : "BM_n0YwBbQBv2_Mr8O06",
  "state" : "COMPLETED",
  "provisioning_progress" : "DONE",
  "provision_start_time" : 1704328720539,
  "provision_end_time" : 1704328721409,
  "resources_created" : [
    {
      "workflow_step_name" : "create_connector",
      "workflow_step_id" : "workflow_step_1",
      "resource_type" : "connector_id",
      "resource_id" : "Bc_o0YwBbQBv2_Mrce3E"
    },
    {
      "workflow_step_name" : "register_remote_model",
      "workflow_step_id" : "workflow_step_2",
      "resource_type" : "model_id",
      "resource_id" : "CM_o0YwBbQBv2_Mrc-2e"
    },
    {
      "workflow_step_name" : "deploy_model",
      "workflow_step_id" : "workflow_step_2-deploy",
      "resource_type" : "model_id",
      "resource_id" : "CM_o0YwBbQBv2_Mrc-2e"
    }
  ]
}
```

Reply:
Ah, I see! This should be fine unless someone has similar steps named "foo" and "foo-deploy" :)